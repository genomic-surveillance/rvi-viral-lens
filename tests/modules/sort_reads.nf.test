nextflow_process {

    name "Test Process sort_reads_with_krakentools"
    script "modules/sort_reads.nf"
    process "sort_reads_with_krakentools"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
                test_kraken_loc = "${projectDir}/tests/test_data/test_kraken_outputs"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = tuple(
                    params.meta, 
                    file("${params.test_kraken_loc}/test.kraken.output"),
                    [
                        file("${params.test_kraken_loc}/test.class_seqs_1.fq"),
                        file("${params.test_kraken_loc}/test.class_seqs_2.fq")
                    ],
                    file("${params.test_kraken_loc}/test.report.txt"))
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
