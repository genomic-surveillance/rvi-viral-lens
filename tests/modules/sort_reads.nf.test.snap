{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.11320.extracted_1.fq:md5,56276303ada6cd6d1d27acf597d65050",
                            "test.11320.extracted_2.fq:md5,3377d5f5d6c3bf86b1ec40cf8cca2ba7",
                            "test.2697049.extracted_1.fq:md5,b0eaac69db411379704837a2b2a9dbd8",
                            "test.2697049.extracted_2.fq:md5,705ce6f753733d8ba3ccdffad193ef52",
                            "test.9000016.extracted_1.fq:md5,a133e4d55617e4847328e4604da16588",
                            "test.9000016.extracted_2.fq:md5,28fff872af8e9100aef79cf76249227d",
                            "test.9000017.extracted_1.fq:md5,3faf1b87cd995ba4d44e7c5c2e1d4f64",
                            "test.9000017.extracted_2.fq:md5,1da37f9acad03d4afb312cbf44345303",
                            "test.9000018.extracted_1.fq:md5,e95a61a8e9d0fbecacfbedb49709c1ec",
                            "test.9000018.extracted_2.fq:md5,c0e9c5d21b063ac0b5a60d1f8fd49db0",
                            "test.9000019.extracted_1.fq:md5,5396e8d5be49befa7eba0baffc50b925",
                            "test.9000019.extracted_2.fq:md5,a73a4c7e92707de0ed865542461014de",
                            "test.9000020.extracted_1.fq:md5,42f3c1d174b1f98545632911a9bc8bac",
                            "test.9000020.extracted_2.fq:md5,c7cd2c6b45819395897963631b1d60bc",
                            "test.9000021.extracted_1.fq:md5,8edc229a2df2a5e6efe5a7f20b597da4",
                            "test.9000021.extracted_2.fq:md5,2dc28e5e60df5741bd4ad3bb9acf1399",
                            "test.9000022.extracted_1.fq:md5,376c188d027a23d9a1504aabc2e48a7e",
                            "test.9000022.extracted_2.fq:md5,d2caea79eb9b2a968b09d90ba2ab4a6a",
                            "test.9000023.extracted_1.fq:md5,415531fd6a88deabb8a61bdcf91cc037",
                            "test.9000023.extracted_2.fq:md5,0a2001d579ab1c2032528bf6e052b089"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "22.04.5"
        },
        "timestamp": "2024-02-28T15:16:43.733364"
    }
}